apps/api/v2/src/modules/organizations/repositories/organizations-membership.repository.ts

```diff
@@ -0,0 +1,72 @@
+import { CreateOrgMembershipDto } from "@/modules/organizations/inputs/create-organization-membership.input";
+import { PrismaReadService } from "@/modules/prisma/prisma-read.service";
+import { PrismaWriteService } from "@/modules/prisma/prisma-write.service";
+import { Injectable } from "@nestjs/common";
+
+@Injectable()
+export class OrganizationsMembershipRepository {
+  constructor(private readonly dbRead: PrismaReadService, private readonly dbWrite: PrismaWriteService) {}
+
+  private async findAllTeamsInOrg(organizationId: number) {
```

Comment:
I think this belongs in `organizations-teams.repository.ts`

Reply:
I see `findAllTeamsInOrg` is used in this repository below - I would dependency inject `organizations-teams.repository.ts` into the organizations membersip service, which gets IDs of all teams. And then they are passed as parameters to this repository function instead of organizationId e.g.:

```
 async findOrgMembershipsPaginated(organizationTeamsIds: number[], skip: number, take: number) {
    return this.dbRead.prisma.membership.findMany({
      where: {
        teamId: { in: organizationTeamsIds },
      },
      skip,
      take,
    });
  }
```